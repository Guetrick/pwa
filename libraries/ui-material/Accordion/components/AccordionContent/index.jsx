import React, { useRef, useEffect } from 'react';
import PropTypes from 'prop-types';
import { useSpring, animated } from 'react-spring';
import * as styles from './style';

/**
 * The accordion content component.
 * @param {Object} props The component props.
 * @returns {JSX}
 */
function AccordionContent({ children, open }) {
  const ref = useRef(null);
  const height = (ref.current === null) ? 'auto' : ref.current.clientHeight;
  const [style, set] = useSpring(() => ({
    height: !open ? 0 : height,
    config: { duration: 150 },
  }));

  // Change the styles when the 'open' flag changes.
  useEffect(() => {
    set({ height: !open ? 0 : height });
  }, [open]);

  return (
    <animated.div className={styles.content} style={style}>
      <div ref={ref} className={styles.contentInner}>
        {children}
      </div>
    </animated.div>
  );
}

AccordionContent.propTypes = {
  children: PropTypes.node.isRequired,
  open: PropTypes.bool,
};

AccordionContent.defaultProps = {
  open: false,
};

export default AccordionContent;
