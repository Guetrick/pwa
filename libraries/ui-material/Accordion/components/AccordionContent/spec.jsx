import React from 'react';
import { shallow } from 'enzyme';
import AccordionContent from './index';

describe('<AccordionContent />', () => {
  it('should render as closed', () => {
    const wrapper = shallow((
      <AccordionContent>
        <div id="test">Some Child</div>
      </AccordionContent>
    ));

    expect(wrapper.find('#test').text()).toEqual('Some Child');
    expect(wrapper.prop('style').height.value).toEqual(0);
  });

  it('should render as open', () => {
    const wrapper = shallow((
      <AccordionContent open>
        <div id="test">Some Child</div>
      </AccordionContent>
    ));

    expect(wrapper.find('#test').text()).toEqual('Some Child');
    expect(wrapper.prop('style').height.value).toEqual('auto');
  });
});
