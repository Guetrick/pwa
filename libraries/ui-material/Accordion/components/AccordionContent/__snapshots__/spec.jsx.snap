// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AccordionContent /> should render as closed 1`] = `
<section
  className={
    Object {
      "data-css-rdsogp": "",
    }
  }
  style={
    Object {
      "height": 0,
    }
  }
>
  <div
    className={
      Object {
        "data-css-yh2lvw": "",
      }
    }
  >
    <div>
      Some Child
    </div>
  </div>
</section>
`;

exports[`<AccordionContent /> should render as open 1`] = `
<section
  className={
    Object {
      "data-css-rdsogp": "",
    }
  }
  style={
    Object {
      "height": 0,
    }
  }
>
  <div
    className={
      Object {
        "data-css-yh2lvw": "",
      }
    }
  >
    <div>
      Some Child
    </div>
  </div>
</section>
`;
